src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -572,6 +481,131 @@ public static List<String> getJacocoAgentOptions() {
         };
     }
 
+    /**
+     * Initializes {@code JENKINS_HOME}, but does not start Jenkins.
+     *
+     * This method does not need to be invoked when using {@code @Rule} or {@code @ClassRule} to run {@code RealJenkinsRule}.
+     */
+    @SuppressFBWarnings(value = "PATH_TRAVERSAL_IN", justification = "irrelevant")
+    public void provision(Description description) throws Exception {
+        this.description = description;
+        if (home.get() != null) {
+            throw new IllegalStateException(description + " was already provisioned");
+        }
+        if (war == null) {
+            war = findJenkinsWar();
+        }
+        home.set(tmp.allocate());
+        LocalData localData = description.getAnnotation(LocalData.class);
+        if (localData != null) {
+            new HudsonHomeLoader.Local(description.getTestClass().getMethod(description.getMethodName()), localData.value()).copy(getHome());
+        }
+        File plugins = new File(getHome(), "plugins");
+        Files.createDirectories(plugins.toPath());
```

Comment:
This line was changed to use NIO to get rid of a SpotBugs warning. Not sure why it wasn't erroring before.

Reply:
Is that the only change being made as this code is moved out of `apply`? The way the diff is structured it is very hard to verify, unfortunately. (Sometimes it is possible to arrange new methods in such a way that `diff -w` is tricked into showing only small hunks, but I cannot see that working here.)

Reply:
Ah never mind! It does work here if you just hide whitespace: https://github.com/jenkinsci/jenkins-test-harness/pull/690/files?diff=unified&w=1